### DISCUSSION TITLE
Any way to run the large model with 6 GB of VRAM? (Probably not possible, but doesn't hurt to ask!)

### DISCUSSION THREAD

**Discussion by athu16** [openai/whisper#180] [Q&A] [answered]
I've been translating a bunch of audio from Marathi to English, and there's a large jump in quality from the Medium model to the Large model. I've been using Google Colab to run the large model, but it would be really convenient if I could run it locally, even if it means compromising some speed.

I've been looking at the rapid development of Stable Diffusion (and DreamBooth), and people have managed to drastically reduce VRAM requirements using methods like 16-bit / 8-bit precision, splitting the model into chunks which are loaded into VRAM as required, etc.

So I was wondering if someone had tried doing a similar thing with Whisper. I understand both models are completely different from each other, so you probably can't use the same techniques on both, but I thought I'd ask anyway.

**Comment by jongwook** [comment, marked as answer]
Could be possible with the techniques you mentioned. One simple method could be that, since the model is half encoder and half decoder, you could encode the audio features, unload the encoder, and then load the decoder part to do the decoding.

Disabling beam search (in favor of greedy decoding which is faster but more prone to repetition looping) and key-value caching (which you can find in the class `PyTorchInference`) could also marginally reduce the VRAM usage.

**Reply by jongwook** [reply]
For visibility: I have an example implementation of a similar idea to this using 2 GPUs with smaller VRAM in https://github.com/openai/whisper/discussions/360#discussioncomment-4156445

**Comment by ProjectEGU** [comment]
If you succeed in loading the large model with 6GB VRAM, can you please make a PR? I'm sure it would help many people who don't have enough memory to run the large model but don't want to upgrade their GPU just for more VRAM.

**Comment by FurkanGozukara** [comment]
yes just due to VRAM requirement of Whisper, I have purchased a new card RTX 3060. Here even I made a speed test video : 

How Good is RTX 3060 for ML AI Deep Learning Tasks and Comparison With GTX 1050 Ti and i7 10700F CPU
https://www.youtube.com/watch?v=q8Q8CCDdSKo

**Comment by phineas-pta** [comment]
take a look at `faster-whisper`: https://github.com/guillaumekln/faster-whisper
